.github/workflows/linting.yml

```diff
@@ -0,0 +1,28 @@
+name: Linting
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+jobs:
+
+  flake8:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install dependencies
+        run: |
+          pip install flake8
+      - name: Run linting
+        run: |
+          flake8 .
```

Comment:
what does flake8 do here? afaik it flags non-PEP8-conforming bits of code, but they are not fixed right? if that is so, how does that help if automated in the workflow?

Reply:
I think you are right I added a new tool using the official [blacken workflow integration](https://black.readthedocs.io/en/stable/integrations/github_actions.html)

Reply:
I just changed this to use black, following their [official instructions](https://black.readthedocs.io/en/stable/integrations/github_actions.html).

Reply:
according to your links ```Please note that the [--check flag](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#labels-exit-code) is required so that the workflow fails if Black finds files that need to be formatted.```

So the idea is that we blacken the code before merging and then this just checks whether it's done?

Reply:
@manoskary: We both did the same at roughly the same time! I tried to merge, but you beat me to it 🤣

Reply:
looks good to me.